jovian/utils/clone.py

```diff
@@ -71,7 +71,15 @@ def clone(slug, version=None, fresh=True, include_outputs=True, overwrite=False)
     # Download gist metadata
     ver_str = '(version ' + str(version) + ')' if version else ''
     log('Fetching ' + slug + " " + ver_str + "..")
-    gist = get_gist(slug, version, fresh)
+    
+    # Try to get gist with current user 
+    try:
+        gist = get_gist(slug, version, fresh)
+    except Exception as exception:
+        log_msg = str(exception)
+        log(log_msg)
+        return
+
```

Comment:
![LIbrary_test](https://user-images.githubusercontent.com/42057645/93077208-1ffe8780-f6a6-11ea-867c-f225a4fabf93.png)

Reply:
<h1>Multiple Solution</h1> 

1. Change the `get_gist(slug, version, fresh)` in clone.py.  we can change 
 `raise Exception('Failed to retrieve Gist: ' + pretty(res))`   To <br>`log('Failed  to retrieve Gist: ' + pretty(res))`
2. First check the user login using `creds_exist()` then make a network call with `gist-access` end-point then proceed
<h4> To get the required error log </h4>
 Can change the error message in backend for a library call OR simply based on the response error message from backend we change that message.

Reply:
You can have first check for `creds_exist()` and log `......jovian.configure`. Since this information is available in library and also dependent on library (logged in state is basically when you have configured your API key).
Not having this check will make get request twice for a case where
1. User has not yet configure and
2. Is not a collaborator/owner to that gist

For the second log `....as a collaborator` You can make a call to `/gist-access` and log the MSG. 
You should not have checks based on error/response message strings, in a case where if they are changed this would break its functionality.

Reply:
For the Public repository too we need to log in?